common/json/json.cpp

```diff
@@ -147,6 +147,37 @@ ExpectedKeyValueMap ToKeyValuesMap(const json::Json &j) {
 	return kv_map;
 }
 
+template <typename T>
+expected::expected<T, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok) {
+	auto exp_value = json.Get(key);
+	if (!exp_value) {
+		if (missing_ok == MissingOk::Yes
+			&& exp_value.error().code != json::MakeError(json::KeyError, "").code) {
+			return T();
+		} else {
+			auto err = exp_value.error();
+			err.message += ": Could not get `" + key + "` from state data";
+			return expected::unexpected(err);
+		}
+	}
+	return exp_value.value().Get<T>();
+}
+// The number of instantiations is pretty much set in stone since it depends on the number of JSON
+// types, which isn't going to change. So use explicit instantiation for compile time efficiency.
+template expected::expected<KeyValueMap, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+template expected::expected<vector<string>, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+template expected::expected<string, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+template expected::expected<int64_t, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+template expected::expected<double, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+template expected::expected<bool, error::Error> Get(
+	const json::Json &json, const string &key, MissingOk missing_ok);
+
```

Comment:
Just for a confirmation/clarification -- all these above are so that you can write `Get<string>()` instead of `GetString()`? I feel like I may be missing the point here.

Reply:
Uhm, no..? There is exactly one implementation, above these six instantiations.